### PR TITLE
Rabbitmqsource schema

### PR DESCRIPTION
**Pull request by ChunyiLyu**
# Changes

- :broom: after [this PR to pkg](https://github.com/knative/pkg/pull/2431), we can use controller-gen to generate CRD `rabbitmqsources`
- `make manifests` target to generate CRD `rabbitmqsources`
-  :broom: renaming for `rabbitmqsource.spec` fields:
spec.exchange_config -> spec.exchangeConfig
spec.queue_config -> spec.QueueConfig
spec.exchange_config.auto_deleted -> spec.exchangeConfig.autoDelete
spec.queue_config.delete_when_unused -> spec.QueueConfig.autoDelete
spec.queue_config.routing_key -> spec.QueueConfig.routingKey
spec.channel_config -> spec.channelConfig
spec.channel_config.global_qos -> spec.channelConfig.globalQos
spec.channel_config.prefetch_count -> spec.channelConfig.prefetchCount

/kind cleanup

### CODE REVIEW

File: config/source/300-rabbitmqsource.yaml [hunk 1]
```diff
@@ -1,51 +1,290 @@
...
+    categories:
+    - all
+    - knative
+    - eventing
+    - sources
+    - importers
+    kind: RabbitmqSource
+    listKind: RabbitmqSourceList
+    plural: rabbitmqsources
+    singular: rabbitmqsource
```
Comment (salaboy) [inline config/source/300-rabbitmqsource.yaml:22]: @ChunyiLyu can we define a `shortName` for this resources?

File: config/source/300-rabbitmqsource.yaml [hunk 2]
```diff
@@ -1,51 +1,290 @@
...
-    categories:
-      - all
-      - knative
-      - eventing
-      - sources
-      - importers
-    kind: RabbitmqSource
-    plural: rabbitmqsources
-  scope: Namespaced
+status:
```
Comment (salaboy) [inline config/source/300-rabbitmqsource.yaml:285]: @ChunyiLyu do we need this status?

**Comment by salaboy** [comment]
@ChunyiLyu and conformance tests are failing due timeouts.. in knative eventing as well.. so it is ok to wait for those to get resolved and then retest here

File: config/source/300-rabbitmqsource.yaml (hunk 2 above)
Comment (ChunyiLyu) [reply config/source/300-rabbitmqsource.yaml:285]: Created an issue for it already: https://github.com/knative-sandbox/eventing-rabbitmq/issues/593
This PR will focus on migrating to controller-gen to keep it concise and ideally merged quickly.

File: config/source/300-rabbitmqsource.yaml (hunk 1 above)
Comment (ChunyiLyu) [reply config/source/300-rabbitmqsource.yaml:22]: Sure we can. Any suggestions?

**Comment by ChunyiLyu** [comment]
@salaboy No. The script succeeds in main branch and other PRs, so it's definitely something I did here broke it. It’s unclear to me why that’s the case still.

File: config/source/300-rabbitmqsource.yaml (hunk 1 above)
Comment (ChunyiLyu) [reply config/source/300-rabbitmqsource.yaml:23]: Add short name `rmqsource` per suggestion from team slack: https://vmware.slack.com/archives/C02BPEY4FU2/p1646826542641919

File: source/source-example.yaml [hunk 3]
```diff
@@ -14,20 +14,20 @@ spec:
...
-    routing_key: RABBITMQ_ROUTING_KEY
-    durable: RABBITMQ_QUEUE_CONFIG_DURABLE
-    delete_when_unused: RABBITMQ_QUEUE_CONFIG_AUTO_DELETED
-    exclusive: RABBITMQ_QUEUE_CONFIG_EXCLUSIVE
-    nowait: RABBITMQ_QUEUE_CONFIG_NOWAIT
+  exchangeConfig:
+    name: RABBITMQ_exchangeConfig_NAME
+    type: RABBITMQ_exchangeConfig_TYPE
+    durable: RABBITMQ_exchangeConfig_DURABLE
+    autoDelete: RABBITMQ_exchangeConfig_AUTO_DELETED
```
Comment (gab-satchi) [inline source/source-example.yaml:21]: is this block intentional? looks like a victim of search n replace

Comment (ChunyiLyu) [reply source/source-example.yaml]: My bad. Updated

File: config/source/300-rabbitmqsource.yaml [hunk 4]
```diff
@@ -1,51 +1,291 @@
...
+    - all
+    - knative
+    - eventing
+    - sources
+    - importers
+    kind: RabbitmqSource
+    listKind: RabbitmqSourceList
+    plural: rabbitmqsources
+    shortNames:
+    - rmqsource
```
Comment (salaboy) [inline config/source/300-rabbitmqsource.yaml:22]: 👍

**Pull request merged**